community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryPropertyChecker.java

```diff
@@ -0,0 +1,312 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.neo4j.consistency.checking.full;
+
+import java.util.Arrays;
+import java.util.Iterator;
+
+import org.neo4j.collection.primitive.Primitive;
+import org.neo4j.collection.primitive.PrimitiveIntIterator;
+import org.neo4j.collection.primitive.PrimitiveIntObjectMap;
+import org.neo4j.collection.primitive.PrimitiveIntObjectVisitor;
+import org.neo4j.collection.primitive.PrimitiveIntSet;
+import org.neo4j.consistency.checking.CheckDecorator;
+import org.neo4j.consistency.checking.CheckerEngine;
+import org.neo4j.consistency.checking.ComparativeRecordChecker;
+import org.neo4j.consistency.checking.OwningRecordCheck;
+import org.neo4j.consistency.report.ConsistencyReport;
+import org.neo4j.consistency.store.DiffRecordAccess;
+import org.neo4j.consistency.store.RecordAccess;
+import org.neo4j.kernel.impl.store.MandatoryPropertyConstraintRule;
+import org.neo4j.kernel.impl.store.RecordStore;
+import org.neo4j.kernel.impl.store.SchemaStorage;
+import org.neo4j.kernel.impl.store.record.DynamicRecord;
+import org.neo4j.kernel.impl.store.record.NodeRecord;
+import org.neo4j.kernel.impl.store.record.PrimitiveRecord;
+import org.neo4j.kernel.impl.store.record.PropertyBlock;
+import org.neo4j.kernel.impl.store.record.PropertyRecord;
+import org.neo4j.kernel.impl.store.record.Record;
+import org.neo4j.kernel.impl.store.record.RelationshipRecord;
+
+import static org.neo4j.consistency.checking.full.NodeLabelReader.getListOfLabels;
+
+public class MandatoryPropertyChecker extends CheckDecorator.Adapter
+{
+    private final PrimitiveIntObjectMap<int[]> nodes = Primitive.intObjectMap();
+    private final PrimitiveIntObjectMap<int[]> relationships = Primitive.intObjectMap();
+
+    public MandatoryPropertyChecker( RecordStore<DynamicRecord> schemaStore )
+    {
+        for ( Iterator<MandatoryPropertyConstraintRule> rules =
+              new SchemaStorage( schemaStore ).schemaRules( MandatoryPropertyConstraintRule.class );
+              safeHasNext( rules ); )
```

Comment:
Look at what `safeHasNext` does...

If the schemaStore is inconsistent, `hasNext` will throw an exception when trying to read the inconsistent record. `safeHasNext` allows us to just continue to the next record at that point.